src/client/cl_entity.c

```diff
@@ -87,23 +96,8 @@ static void Cl_ReadDeltaEntity(cl_frame_t *frame, entity_state_t *from, uint16_t
 	ent->frame_num = cl.frame.frame_num;
 	ent->current = *to;
 
-	// update clipping matrices, snapping the entity to the frame
-	if (ent->current.solid) {
-		if (ent->current.solid == SOLID_BSP) {
-			if (bits & (U_ORIGIN | U_ANGLES)) {
-				Matrix4x4_CreateFromEntity(&ent->matrix, ent->current.origin, ent->current.angles, 1.0);
-				Matrix4x4_Invert_Simple(&ent->inverse_matrix, &ent->matrix);
-			}
-		} else { // bounding-box entities
-			if (bits & U_ORIGIN) {
-				Matrix4x4_CreateFromEntity(&ent->matrix, ent->current.origin, vec3_origin, 1.0);
-				Matrix4x4_Invert_Simple(&ent->inverse_matrix, &ent->matrix);
-			}
-		}
-	}
-
 	// mark the lighting cache as dirty
-	if (bits & U_ORIGIN) {
+	if (bits & (U_ORIGIN | U_TERMINATION | U_ANGLES | U_MODELS | U_BOUNDS)) {
 		ent->lighting.state = LIGHTING_DIRTY;
 	}
```

Comment:
It's pointless to update lighting every single renderer frame. Updating it every server frame, if something changes, is totally adequate. This should shave the time spent in `Cl_Trace`, and it also tidies up `Cl_Interpolate` below.